### DISCUSSION TITLE
Exit dialog

### DISCUSSION THREAD

**Discussion by ZeljkoBG** [osmandapp/OsmAnd#19229] [Ideas]
Please add "Exit" dialog **Yes/No**
With Android Back button, the application is falling out. And while driving. Very uncomfortable.

**Comment by rysson** [comment]
Nice idea. But only if enabled in settings IMHO.

**Comment by ZeljkoBG** [comment]
Not in settings. It is programmed in the application. Developers don't seem to use OsmAnd so they don't know about this bug.

**Reply by vshcherb** [reply]
Use it very often, we just don't click back. During navigation probably it's could be a good option

**Reply by rysson** [reply]
OK, I could understand @ZeljkoBG problem.
I've check again, there is no dialog now, and it's better option for me. If I want to exit quickly to do something else I don't want to focus on extra dialog. I can re-enter to OsmAnd very easy, and continue navigation.

It's way I propose add to the settings how to react on back button (example):
- show exit dialog
- double back to exit
- exit immediately (default)

I need rather big button "exit" on screen or in actions, because I have new car head unit and back button it really small, I have to find some stronger glasses to see it :-P

**Comment by ZeljkoBG** [comment]
The problem is during navigation. There's another problem. If during the navigation we press accidentally or intentionally android Back navigation runs in the background. I scrolled the screen to stop it several times. All during the ride. Uncomfortable and risky.

**Comment by ZeljkoBG** [comment]
There is no need to discuss how to leave the application. Install any GPS navigation and see the solution.

**Reply by rysson** [reply]
OK, I there is no need to discuss than what I could say – it's everything is OK for me now (immediately exit), nothing to change. 
We could close this topic. Thank you.

**Reply by vshcherb** [reply]
Google Maps - Back exit navigation no confirmation needed

**Comment by ZeljkoBG** [comment]
1. But that's why it doesn't work in the background! No additional scrolling of the Notification bar is needed to stop navigation. Because it works in the background.
2. Google Maps doesn't need anyone's role model.
3. See Sygic, iGO, TomTom,...
